GitCommands/Git/GitModule.cs

```diff
@@ -359,6 +359,11 @@ public string GitCommonDirectory
         {
             get
             {
+                if (string.IsNullOrEmpty(WorkingDir))
+                {
+                    return "";
+                }
```

Comment:
I need a comment here because the relation between `_gitExecutable` and `WorkingDir` is not obvious. Perhaps the property `WorkingDir` can be declared `init;`.

Reply:
If the GitModule has no directory set, the command should not return any path, even if GE has been started from a Git repo.
So not really related to the executable.
This could be applied to other commands too (but not so many commands are executed when the GitModule has no working dir.)
Do you want a comment?

WorkingDir could be init, how could that improve.

Reply:
> Do you want a comment?

Yes, would be good.

> WorkingDir could be init, how could that improve.

Then it will be clearer that `WorkingDir` will not change during runtime. Hence the result of this check will always be the same, i.e. will not have an inconsistent affect on repeated calls of this function.

Reply:
done